x-pack/plugins/reporting/server/plugin.ts

```diff
@@ -49,22 +43,25 @@ export class ReportingPlugin
     });
 
     const { elasticsearch, http } = core;
-    const { licensing, security } = plugins;
+    const { licensing, security, taskManager } = plugins;
     const { initializerContext: initContext, reportingCore } = this;
 
     const router = http.createRouter();
     const basePath = http.basePath.get;
+    const task = new ReportingTask(this.reportingCore, this.logger);
 
     reportingCore.pluginSetup({
       elasticsearch,
       licensing,
       basePath,
       router,
       security,
+      task,
     });
 
     registerReportingUsageCollector(reportingCore, plugins);
     registerRoutes(reportingCore, this.logger);
+    task.register(taskManager);
```

Comment:
Getting this figured out was a little tricky because it has to be called in sync. I don't think we'd be able to add it as part of background setup. That means at the point of time where we register the task, our Reporting config is not available.